### PR TITLE
Add a helper script for testing GitHub Actions on forked repositories

### PR DESCRIPTION
**Pull request by hhiroshell**
## Summary
The `buildpacks/lifecycle` repository has [a helper script](https://github.com/buildpacks/lifecycle/blob/9f262b3e5dfb6b59d331f6bbf30f506e5bd3a895/DEVELOPMENT.md#testing-github-actions-on-forks) for testing GitHub Actions on forked repositories. This PR adds a script `tools/test-fork.sh` for the same purpose to the `buildpacks/pack`.

The new script makes the following changes to the existing workflows:
- Disable the workflows that should not run on forked repositories by renaming the workflow files
- Change the container registry account and other related information to build and push a different container image from the upstream
- Remove build target architectures that require self-hosted runner
- Remove upstream maintainers from the recipients of the benchmark alert

## Output

N/A

## Documentation

- Should this change be documented?
    - [ ] Yes, see #___
    - [x] No

## Related

This PR is a preparation for making corrections to #2111.

Here is results of workflows tested on my fork repository.
- [benchmark.yml](https://github.com/hhiroshell/pack/actions/runs/10069677310)
- [build.yml](https://github.com/hhiroshell/pack/actions/runs/10070501288)
- [check-latest-release.yml](https://github.com/hhiroshell/pack/actions/runs/10070415681) (The workflow has failed as intended after detecting the CVE and creating an issue.)
- [codeql-analysis.yml](https://github.com/hhiroshell/pack/actions/runs/10069677316)
- [compatibility.yml](https://github.com/hhiroshell/pack/actions/runs/10069677325)
- [delivery-docker.yml (pushing to the index.docker.io)](https://github.com/hhiroshell/pack/actions/runs/10070897920)
- [delivery-docker.yml (pushing to the ghcr.io)](https://github.com/hhiroshell/pack/actions/runs/10072767636)
- [pages-build-deployment.yml](https://github.com/hhiroshell/pack/actions/runs/10069692490)
- [release-merge.yml](https://github.com/hhiroshell/pack/actions/runs/10070501296)

### CODE REVIEW

File: .github/workflows/delivery-docker.yml [hunk 1]
```diff
@@ -16,8 +16,9 @@ on:
         default: false
 
 env:
+  REPO_NAME: 'index.docker.io'
```
Comment (natalieparellano) [inline .github/workflows/delivery-docker.yml:19]:
```suggestion
  REGISTRY_NAME: 'index.docker.io'
```

File: .github/workflows/delivery-docker.yml [hunk 2]
```diff
@@ -52,7 +53,7 @@ jobs:
         with:
           ref: v${{ steps.version.outputs.result }}
       - name: Determine App Name
-        run: 'echo "IMG_NAME=${{ env.USERNAME }}/${{ env.IMG_NAME }}" >> $GITHUB_ENV'
+        run: 'echo "IMG_NAME=${{ env.REPO_NAME }}/${{ env.USER_NAME }}/${{ env.IMG_NAME }}" >> $GITHUB_ENV'
```
Comment (natalieparellano) [inline .github/workflows/delivery-docker.yml:56]:
```suggestion
        run: 'echo "IMG_NAME=${{ env.REGISTRY_NAME }}/${{ env.USER_NAME }}/${{ env.IMG_NAME }}" >> $GITHUB_ENV'
```

**Review by natalieparellano — COMMENTED** [review]
This is awesome, thank you @hhiroshell!

**Comment by hhiroshell** [comment]
@natalieparellano 
Thank you for your review!

The CI task `build / test (windows-lcow) (pull_request) ` is failing. I have no idea what changes might have caused it, but I'm currently investigating it.

**Comment by hhiroshell** [comment]
It seems the task failures are random and unpredictable, and it's difficult for me to identify the cause from the CI logs alone.

Do you have any ideas about what might be causing this? Alternatively, could you please rerun the `build / test (windows-lcow) (pull_request)` task?

**Comment by natalieparellano** [comment]
Thank you @hhiroshell. I wouldn't worry too much about these flakes. We have a number of them that we need to fix (especially on Windows). I re-kicked CI just now, let's see what happens...

**Review by natalieparellano — APPROVED** [review]

**Pull request merged**